src/types/events.ts

```diff
@@ -0,0 +1,35 @@
+
+import { TaskInstance } from "../TaskInstance";
+
+export const EventTargetOptions = {
+    OnError: "onError",
+} as const;
+
+export type EventTarget = typeof EventTargetOptions[keyof typeof EventTargetOptions];
+
+export interface IEventArgs {
+    sender: TaskInstance<any>,
+    params?: any[],
+    error?: unknown,
+    data?: unknown
+}
```

Comment:
Using an interface and the I prefix are just a habit from C# world, I can remove it. Also, it should be a type in this instance since we are the originator of the object shape. Interfaces, iirc, are for when you are the receiver of the object shape, such as the return object of an API.

Reply:
Removed the interfaces entirely as it should have been from the onset.